### ISSUE TITLE
Treat data as THUMB code, not bytes

### ISSUE THREAD

**Issue by Mte90** [arantonitis/luvdis#2]
I am following this tutorial https://web.archive.org/web/20071012203301/http://etk.scener.org/?op=showtutorial&st=2.
I already patched with okteta the original rom using the memory address reference, the problem that I cannot find for `080a9430`.

So to see if this tool works I disassembled the original rom and th emodded one, and did a diff of those files.
![immagine](https://user-images.githubusercontent.com/403283/78160571-72bfec00-7444-11ea-99bb-bb0afff7be5e.png)

The changes are present but not the code itself for those addresses.

**Comment by aarant** [comment]
The disassembler probably wasn't able to determine for certain that that section of the ROM was code and not miscellaneous data. If it cannot determine that a region is code, it just treats it as raw bytes since generally that's more useful than seeing garbage code everywhere.

The function detection isn't perfect yet, so I'd suggest reading the tutorial, going to that address and scrolling up until you see a `push {lr}` of some kind. Then just take the address that is at and put it in a `functions.cfg` file:

`thumb_func 0x<address>`

and pass that file to the disassembler: `luvdis -c functions.cfg ...`

You also might have luck passing the command-line options that loosen the requirements to consider a section of the ROM as code:
```bash
luvdis --min-calls 1 --min-length 1 ...
```

Please let me know if that works!

**Comment by marccane** [comment]
I'm experiencing the same thing. I disassembled almost all of pokemon emerald's .text with  --start 0x8000000 --stop 0x82e6d54, and the function I wanted (08098D10) appeared as raw bytes (there's a lot of raw byte blocks in there).

I tried --no-guess expecting it to disassemble everything but it did the oposite. 
A flag to disassembly everything would be very convenient as I rather prefer it to disassemble data than having to specify each function I want to disassemble in the config file and rerunning.

On another note the function I wanted is referenced at least in 3 call (bl) instructions and even using "--min-calls 1 --min-length 1" as sugested didn't help to "guess it".
